invenio_communities/views/decorators.py

```diff
@@ -67,10 +65,61 @@ def view(**kwargs):
     return view
 
 
+def require_community_read(f):
+    """Decorate a view to require community read permissions for accessing the 
+    view."""
+    @pass_community
+    @wraps(f)
+    def view(community=None, **kwargs):
+        """Raise a PermissionDeniedError if the user doesn't have the appropiate
+        permission."""
+        if community.has_permissions_to(["read"])["can_read"]:
+            return f(**kwargs)
+        else:
+            raise PermissionDeniedError()
+
+    return view
+
+def require_community_search_requests(f):
```

Comment:
This has to be refactored because it adds a decorator `@pass_community` and, where it is used, it also passes the same decorator, 2 times.